src/Security/AzureOidcAuthenticator.php

```diff
@@ -47,27 +59,53 @@ public function authenticate(Request $request): Passport
             // Validate claims
             $claims = $this->validateClaims($request);
 
-            // Extract properties from claims
-            $name = $claims['navn'];
-            $email = $claims['email'];
-            $oidcGroups = $claims['groups'] ?? [];
+            $provider = $claims['open_id_connect_provider'];
+
+            switch ($provider) {
+                case self::OIDC_PROVIDER_EXTERNAL:
+                    $signInName = $claims[$this->oidcExternalClaimId];
+
+                    if (empty($signInName)) {
+                        throw new CustomUserMessageAuthenticationException('Missing required claim '.$this->oidcExternalClaimId);
+                    }
+
+                    $type = UserTypeEnum::OIDC_EXTERNAL;
+                    $name = ExternalUserService::EXTERNAL_USER_DEFAULT_NAME;
+                    $providerId = $this->externalUserService->generatePersonalIdentifierHash($signInName);
+                    $email = null;
```

Comment:
Not sure what the best solution is if the email is not nullable. Either do `$signInName.'@ext'` as a temporary value or make email nullable, but unique. (MySQL allows multiple NULLs in a column with a unique constraint.)

Reply:
Will create a temporary email from providerId when the user has not activated. The mail will be updated after activation.